web/html/src/components/input.js

```diff
@@ -56,20 +59,17 @@ class Form extends React.Component {
 
         const childrenCount = React.Children.count(children);
 
-        if (childrenCount > 1) {
-            return React.Children.map(children, child => this.renderChild(child));
-        } else if (childrenCount === 1) {
-            return this.renderChild(Array.isArray(children) ? children[0] : children);
-        }
+        return React.Children.map(children, child => this.renderChild(child));
     }
 
     renderChild(child) {
         if (typeof child !== 'object' || child === null) {
             return child;
         }
 
-        if (child.type && child.type.prototype !== null &&
-                child.type.prototype instanceof _InputBase) {
+        const fieldTypes = [Text, Password, Check, Select, DateTime];
```

Comment:
Well this seems a bit hammered. The component form shouldn't be aware of the components types we have.

There is an ideal solution for this but it depends on React 16.3 : https://reactjs.org/docs/context.html 

We could have a formPropsProvider and implement a formPropsConsumer on inputBase :) 

I would leave it like this for now and add a comment:

// [LuNeves] [Cedric] After upgrading react to v16 this could be improved with Context: formPropsProvider and a formPropsConsumer on the component inputBase

Reply:
OK will do